### PR TITLE
Add test-coverage for SequenceTemplateAdminService

### PR DESCRIPTION
**Pull request by himasha**
## Purpose
Add test-coverage for SequenceTemplate based admin service operations.

## Approach
Included test-cases for following scenarios.
1. Add/get/update/delete sequence template.
2. Add/get/update dynamic sequence template.
3. Exception handling for invalid scenarios.
4. Enable/disable statistics/tracing for sequence templates.
5. Access count for deployed sequence templates.

### CODE REVIEW

File: integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    }
+
+    @Test(groups = { "wso2.esb" },
+          description = "Test template creation service",
+          priority = 1)
+    public void testCreateTemplate() throws Exception {
+        int prevCount = templateAdmin.getTemplatesCount();
+        templateAdmin.addSequenceTemplate(AXIOMUtil.stringToOM(template1));
+        int latestCount = templateAdmin.getTemplatesCount();
+        org.testng.Assert.assertTrue(prevCount < latestCount, "New template not added");
```
Comment (manoramahp) [inline integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java]: Do we need "org.testng.Assert.assertTrue" here, as we have already imported the class?

File: integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    }
+
+    @Test(groups = { "wso2.esb" },
+          description = "Test dynamic template creation service",
+          priority = 10)
+    public void testCreateDynamicTemplate() throws Exception {
+        int prevCount = templateAdmin.getDynamicTemplateCount();
+        templateAdmin.addDynamicSequenceTemplate(KEY,AXIOMUtil.stringToOM(template1));
+        int latestCount = templateAdmin.getDynamicTemplateCount();
+        org.testng.Assert.assertTrue(prevCount < latestCount, "Dynamic template not added");
```
Comment (manoramahp) [inline integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java]: Similar as above...

File: integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java (hunk 1 above)
Comment (himasha) [reply integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java]: This was added through auto import. Will fix it (y)

Comment (himasha) [reply integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java]: Fixed 5e7fb91734735c829b3ea3c5a2df7acf91b2e166

File: integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java (hunk 2 above)
Comment (himasha) [reply integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java]: Fixed 5e7fb91734735c829b3ea3c5a2df7acf91b2e166

File: integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        OMElement elm = endpointAdmin.getEndpointTemplate(endpointName);
+        Assert.assertNotNull(elm, "Unable to get requested endpoint template");
+        Assert.assertTrue(elm.toString().contains(endpointName), "New endpoint template not listed");
+    }
+
+    @Test(groups = { "wso2.esb" },
+          description = "Test endpoint template update service",
+          priority = 15)
+    public void testUpdateEndpointTemplate() throws Exception {
+        if (!endpointAdmin.hasDuplicateEndpointTemplate(endpointTmpl1)) {
```
Comment (milindaperera) [inline integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java]: if endpointAdmin.hasDuplicateEndpointTemplate(endpointTmpl1) returns true, is it a failure or success?
That path is not asserted here.

Comment (himasha) [reply integration/mediation-tests/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/api/TemplateAdminService.java]: Fixed with d2d9b2c

**Pull request merged**